src/core/patterns.js

```diff
@@ -59,3 +59,61 @@ foam.CLASS({
     }
   ]
 });
+
+/** Causes an class to pool its instances. create() will pull from the pool,
+ and destroy() will return instances to the pool. Object pools can be found
+ in <code>foam.__objectPools__</code>. */
+foam.CLASS({
+  package: 'foam.pattern',
+  name: 'Pooled',
+  axioms: [ foam.pattern.Singleton.create() ],
+
+  methods: [
+    function installInClass(cls) {
+      // Keeping the object pools in an accessible location allows them
+      // to be cleared out.
+      if ( ! foam.__objectPools__ ) { foam.__objectPools__ = {}; }
```

Comment:
Would it make sense to store the objectPool one the class itself?  This would make it possible to garbage collect a class if all the references to it and its instances are removed, regardless of if every instance was .destryoed().

Plus then it removes an extra lookup during create.